src/main/java/com/fryrank/dal/ReviewDALImpl.java

```diff
@@ -36,20 +36,33 @@ public GetAllReviewsOutput getAllReviewsByRestaurantId(@NonNull final String res
         query.addCriteria(equalToRestaurantIdCriteria);
         final List<Review> reviews = mongoTemplate.find(query, Review.class);
 
-        final MatchOperation filterToRestaurantId = match(equalToRestaurantIdCriteria);
-        final GroupOperation averageScore = group(RESTAURANT_ID_KEY).avg("score").as("avgScore");
-        final Aggregation aggregation = newAggregation(filterToRestaurantId, averageScore);
-        final AggregationResults<RestaurantAvgScore> result = mongoTemplate.aggregate(aggregation, "review", RestaurantAvgScore.class);
-        final Optional<RestaurantAvgScore> uniqueResult = Optional.ofNullable(result.getUniqueMappedResult());
+        return new GetAllReviewsOutput(reviews);
+    }
+
+    @Override
+    public GetAggregateReviewInformationOutput getAggregateReviewInformationForRestaurants(@NonNull final List<String> restaurantIds, @NonNull final AggregateReviewFilter aggregateReviewFilter) {
+        final Query query = new Query().addCriteria(Criteria.where(RESTAURANT_ID_KEY).in(restaurantIds));
+
+        Map<String, AggregateReviewInformation> restaurantIdToAggregateReviewInformation = new HashMap<String, AggregateReviewInformation>();
+
+        restaurantIds.stream().forEach(restaurantId -> {
```

Comment:
This makes about 20 calls to MongoDB on a regular restaurants page. We can definitely just make a singular call with 20 restaurant IDs which will save time.

Reply:
Done in [c29a60c](https://github.com/NickPriv/FryRankBackend/pull/54/commits/c29a60cfdd56058b64e9f56ad71e08c02516b820)